pallets/manta-sbt/src/lib.rs

```diff
@@ -434,9 +435,10 @@ pub mod pallet {
             // check that mint type is within time window
             Self::check_mint_time(&chain_info)?;
 
-            let address =
-                Self::verify_eip712_signature(&post.proof, &eth_signature, chain_info.chain_id)
-                    .ok_or(Error::<T>::BadSignature)?;
+            let chain_id = chain_id.unwrap_or(chain_info.chain_id);
```

Comment:
What the difference between `chain_id` and `chain_info.chain_id`?

Reply:
chain_id on the method is as parameters. if not passing chain_id, then use chain_info's chain_id.

Reply:
What if user inputs `chain_id` and `chain_info.chain_id` both, will it affect line 441? I mean will the signature be valid?

Reply:
as long as the chainId when user sign the message match user input's chain id, the signature is valid.

Reply:
The signature is only used for eth address verification. If we set chainId fixed to admin setting. When user change to other network which is almost frequency operation, their signature will be invlid, they'll need to switch back to BSC chain and sign again, this is not a good UE.

Reply:
maybe remove the admin setting, and only accept chain id as parameters, that will not cause any confuse?

Reply:
From a security perspective this is fine as the signature can never be successfully replayed.

Although IMO it is a little weird to allow an signature from ETH chain_id to pass verification for a BSC item